File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i < n; i++) {
            pq.add(Integer.parseInt(br.readLine()));
        }

        long total = 0;

        while (pq.size() > 1) {
            int first = pq.poll();
            int second = pq.poll();

            int sum = first + second;
            total += sum;
            pq.add(sum);
        }

        System.out.print(total);
    }
}

File: 백준/Gold/1715. 카드 정렬하기/카드 정렬하기.kt
import java.util.*

fun main() {
    val n = readln().toInt()
    val heap = PriorityQueue<Int>()
    repeat(n) {
        heap.add(readln().toInt())
    }
    var sum = 0
    repeat(n - 1) {
        var a = heap.peek()
        heap.poll()
        var b = heap.peek()
        heap.poll()
        sum += a + b
        heap.add(a + b)
    }
    println(sum)
}